benchmark/infrastructure/BenchmarkMeasurementContainer.h

```diff
@@ -343,13 +357,24 @@ class ResultGroup : public BenchmarkMetadataGetter {
                         const std::vector<std::string>& rowNames,
                         const std::vector<std::string>& columnNames);
 
+  /*
+  Delete the given `ResultTable` from the group. Note: Because the group has
+  ownership of all contained entries, this will delete the `ResultTable`.
+  */
+  void deleteTable(ResultTable& table);
+
   // User defined conversion to `std::string`.
   explicit operator std::string() const;
   friend std::ostream& operator<<(std::ostream& os,
                                   const ResultGroup& resultGroup);
 
   // JSON serialization.
   friend void to_json(nlohmann::json& j, const ResultGroup& resultGroup);
+
+ private:
+  // The implementation for the general deletion of entries.
+  template <ad_utility::isTypeAnyOf<ResultEntry, ResultTable> T>
+  requires std::same_as<std::decay_t<T>, T> void deleteEntryImpl(T& entry);
```

Comment:
This requires clause is completely reduntant because of your chosen concept (or does `isTypeAnyOf` allow `const ResultEntry&`, which it shouldn't?

Reply:
`isTypeAnyOf` compares types with `isSimilar`, so it would allow `const ResultEntry&`. Which I do not want.

I mean, because of shallow constness, it would still work, but I really do not want that misunderstanding.

Reply:
I understand. Then the misunderstanding was that `isTypeAnyOf` is maybe a little misnamed (but nothing for this PR).
So then you can simply use `std::is_object_v<T>` as your requires clause, which is a bit more idiomatic and has the same effect (no references allowed).

Reply:
No, no, there has been a slight misunderstanding. It can neither be a reference, nor `const`. `std::is_object_v` allows `const`, so it's not enough.